server/game/gameSystems/LookMoverDeckCardsTopOrBottomSystem.ts

```diff
@@ -23,59 +20,67 @@ export class LookMoveDeckCardsTopOrBottomSystem<TContext extends AbilityContext
-                                target: card,
-                                bottom: false,
-                                destination: Location.Deck
-                            }).generateEvent(context.target, context));
-                            // get rid of the card from cards
-                            cards = cards.filter((a) => a !== card);
-                            choiceHandler(context.player);
-                        }),
-                        (() => {
-                            events.push(moveCard({
-                                target: card,
-                                bottom: true,
-                                destination: Location.Deck
-                            }).generateEvent(context.target, context));
-                            // get rid of the card from cards
-                            cards = cards.filter((a) => a !== card);
-                            choiceHandler(context.player);
-                        })]).flat()
-                    });
-                } else {
+            // Each card has two options to be put on top or on bottom for each option we have a handler whcih
+            // recursively calls the function and removes handlers from the list until the card pool reaches 0.
+            const choiceHandler = (player, cards: any[]) => {
+                if (cards.length === 0) {
                     return true;
                 }
-            };
+                // setup the choices for each card top and bottom
+                const handlerChoices = cards.map((card: Card) => [
+                    'Put ' + card.title + ' on top',
+                    'Put ' + card.title + ' on bottom',
+                ]).flat();
 
-            choiceHandler(context.player);
+                context.game.promptWithHandlerMenu(player, {
+                    activePromptTitle: 'Select card to move to the top or bottom of the deck',
+                    choices: handlerChoices,
+                    handlers: cards.map((card: Card) => [(() => {
+                        this.pushMoveEvent(false, card, cards, events, context, choiceHandler);
+                    }),
+                    (() => {
+                        this.pushMoveEvent(true, card, cards, events, context, choiceHandler);
+                    })]).flat()
+                });
+            };
+            choiceHandler(context.player, cards);
         }
     }
 
+    // Helper method for pushing the move card event into the events array.
+    private pushMoveEvent(bottom: boolean, card: Card, cards: any[], events: GameEvent[], context: TContext,
+        choiceHandler: (player: any, cards: any[]) => boolean) {
+        // create a new card event
+        const moveCardEvent = new MoveCardSystem({
+            target: card,
+            bottom: bottom,
+            destination: Location.Deck
+        }).generateEvent(context.target, context);
+        events.push(moveCardEvent);
+        // get rid of the card from cards
```

Comment:
nit: whitespace in here to break these blocks up a bit. generally if it's big enough to require a comment it should get its own code block

Reply:
yessir!